java-frontend/src/main/java/org/sonar/java/resolve/TypeAndReferenceSolver.java

```diff
@@ -933,7 +940,21 @@ public void visitMethodReference(MethodReferenceTree methodReferenceTree) {
         .map(samMethod -> (JavaType) samMethod.returnType().type())
         .ifPresent(samReturnType -> {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this lambda expression number of lines from 24 to at most 20. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1188)